src/CMakeLists.txt

```diff
@@ -0,0 +1,147 @@
+    set(renamed_parser_header ${CMAKE_CURRENT_BINARY_DIR}/${name}_y.tab.h)
+    add_custom_command(OUTPUT ${renamed_parser_header}
+        COMMAND ${CMAKE_COMMAND} -E copy ${BISON_parser_OUTPUT_HEADER} ${renamed_parser_header}
+        MAIN_DEPENDENCY ${BISON_parser_OUTPUT_HEADER}
+    )
+    list(REMOVE_ITEM BISON_parser_OUTPUTS ${BISON_parser_OUTPUT_HEADER})
+    list(APPEND BISON_parser_OUTPUTS ${renamed_parser_header})
+endmacro(generic_bison)
+
+#   Add a flex target named 'scanner'
+macro(generic_flex name)
+    flex_target(
+        scanner
+        ${CMAKE_CURRENT_SOURCE_DIR}/scanner.l
+        ${CMAKE_CURRENT_BINARY_DIR}/${name}_lex.yy.cpp
+        COMPILE_FLAGS -Pyy${name}
+    )
+endmacro(generic_flex)
+
+#   Set the public include locations for a target.
+macro(generic_includes name)
+    target_include_directories(${name}
+        PUBLIC
+        ${CBMC_BINARY_DIR}
+        ${CBMC_SOURCE_DIR}
+        ${CMAKE_CURRENT_BINARY_DIR}
+        ${CMAKE_CURRENT_SOURCE_DIR}
+    )
+endmacro(generic_includes)
+
+#   Link optional modules.
+#   Target is the name of the target with optional components.
+#   Name is the name of the optional target.
+#   Also adds a compile flag signalling to the preprocessor that the library is
+#   used.
+macro(add_if_library target name)
+    if(TARGET ${name})
+        target_link_libraries(${target} ${name})
+
+        string(TOUPPER ${name} upper_name)
+        string(REGEX REPLACE "-" "_" define ${upper_name})
+
+        target_compile_definitions(${target} PUBLIC HAVE_${define})
+    endif()
+endmacro(add_if_library)
+
+#   EXTERNAL PROJECTS
+include(ExternalProject)
+set(extern_location ${CMAKE_CURRENT_BINARY_DIR}/extern)
+
+set(extern_include_directory ${extern_location}/include)
+file(MAKE_DIRECTORY ${extern_include_directory})
+
+set(minisat_lib ${extern_location}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}minisat2-condensed${CMAKE_STATIC_LIBRARY_SUFFIX})
+
+#   minisat download
+#   This downloads minisat2, then patches it.
+#   Then, it injects a minimal CMakeLists.txt so that we can build just the bits
+#   we actually want, without having to update the provided makefile.
+ExternalProject_Add(minisat2-extern
+    PREFIX ${extern_location}
+    URL http://ftp.debian.org/debian/pool/main/m/minisat2/minisat2_2.2.1.orig.tar.gz
+    PATCH_COMMAND patch -p1 -i ${CMAKE_CURRENT_SOURCE_DIR}/../scripts/minisat-2.2.1-patch
+          COMMAND cmake -E copy ${CMAKE_CURRENT_SOURCE_DIR}/../scripts/minisat2_CMakeLists.txt CMakeLists.txt
+          CMAKE_ARGS -DCBMC_INCLUDE_DIR:path=${CMAKE_CURRENT_SOURCE_DIR} -DCMAKE_INSTALL_PREFIX:PATH=<INSTALL_DIR>
+    BUILD_BYPRODUCTS ${minisat_lib}
+)
+
+add_library(minisat2-condensed STATIC IMPORTED)
+set_target_properties(minisat2-condensed PROPERTIES
+    IMPORTED_LOCATION ${minisat_lib}
+    INTERFACE_INCLUDE_DIRECTORIES "${extern_include_directory}"
+)
+add_dependencies(minisat2-condensed minisat2-extern)
+
+#   Override add_executable to automatically sign the target on OSX.
```

Comment:
Where are universal binaries and the MacOS version implemented?

Reply:
That's handled automatically. By default, the build will be for the architecture of the machine running the build. However, you can also supply the flag `-DCMAKE_OSX_ARCHITECTURES=i386;x86_64` to build fat binaries.

Reply:
Will `-mmacosx-version-min=10.9` also be supported? Note that this ought to be the default as we have had problems in the past with users being unable to run the tool on those older MacOS versions.

Essentially, whatever is currently configured in `common` and `config.inc` ought to be the default for the CMake builds as well.